relay/README.md

```diff
@@ -84,91 +65,189 @@ Scene 2:
-  - Listen for connections relayed through QmRelay.
+
+Specify a relay:
+
+- `/p2p-circuit/p2p/QmRelay/p2p/QmTwo`
+  - Dial QmTwo, through QmRelay.
+  - Use peer routing to find an address for QmRelay.
+  - The relay node will also use peer routing, to find an address for QmTwo.
+- `/p2p-circuit/ip4/../tcp/../p2p/QmRelay/p2p/QmTwo`
+  - Dial QmTwo, through QmRelay.
   - Includes info for connecting to QmRelay.
-  - Also makes QmRelay available for relayed dialing, based on how listeners currently relate to dialers.
-- `/p2p/QmRelay/p2p-circuit`
-  - Same as previous example, but use peer routing to find an address for QmRelay.
+  - The relay node will use peer routing to find an address for QmTwo.
+
+Double relay:
+
 - `/p2p-circuit/p2p/QmTwo/p2p-circuit/p2p/QmThree`
   - Dial QmThree, through a relayed connection to QmTwo.
   - The relay nodes will use peer routing to find an address for QmTwo and QmThree.
   - We'll probably not support nested relayed connections for now, there are edge cases to think of.
-- `/ip4/../tcp/../p2p/QmRelay/p2p-circuit/p2p/QmTwo`
-  - Dial QmTwo, through QmRelay.
+
+--
+
+?? I don't understand the usage of the following:
+
+- `/ip4/../tcp/../p2p/QmRelay/p2p-circuit`
+  - Listen for connections relayed through QmRelay.
   - Includes info for connecting to QmRelay.
-  - The relay node will use peer routing to find an address for QmTwo.
+  - Also makes QmRelay available for relayed dialing, based on how listeners currently relate to dialers.
+- `/p2p/QmRelay/p2p-circuit`
+  - Same as previous example, but use peer routing to find an address for QmRelay.
+
+?? I believe we don't need this one:
 - `/p2p-circuit`
   - Use relay discovery to find a suitable relay node. (Neither specified nor implemented.)
   - Listen for relayed connections.
   - Dial through the discovered relay node for any `/p2p-circuit` multiaddr.
 
-TODO: figure out forced addresses.
-TODO: figure out nested relayed connections.
+TODO: figure out forced addresses. -> what is forced addresses?
 
 ## Wire format
 
-The wire format (or codec) is named `/ipfs/relay/circuit` and is simple.
-A variable-length header consisting of two length-prefixed multiaddrs
-is followed by a bidirectional stream of arbitrary data,
-and the eventual closing of the stream.
+### Overview
+
+#### Setup
+
+Peers involved:
+- A, B, R
+- A wants to connect to B, but needs to relay through R
+
+#### Assumptions
+
+A has connection to R, R has connection to B
+
+#### Process
+
+- A opens new stream `sAR` to R using protocol RELAY
+- A writes Bs multiaddr `/ipfs/QmB` on `sAR`
+- R receives stream `sAR` and reads `/ipfs/QmB` from it.
+- R opens a new stream `sRB` to B using protocol RELAY
+- R writes `/ipfs/QmB` on `sRB`
+- B receives stream `sRB` and reads `/ipfs/QmB` from it.
+- B sees that the multiaddr it read is its own and chooses to handle this stream as an endpoint instead of attempting to relay further
+- TODO: step for R to send back status code to A
```

Comment:
Thoughts on status codes:

* R doesn't send code if the circuit is successful only if there was a failure
* A would get a success or failure code from B once the circuit is established